### DISCUSSION TITLE
Debian install

### DISCUSSION THREAD

**Discussion by jbhopkins** [biosaxs-com/atsas-community#1969] [ATSAS package in general]
Hi,

At my beamline we generally run Debian as our Linux OS of choice. Previously there were Debian installers, is there any plan to return to that? Alternatively, Ubuntu installers generally work on the equivalent flavor of Debian. We're running Debian 12 right now. I tried the Ubuntu 22 installer, but it errored on when checking the distribution version:
`[27513] AutomaticRuntimeDependencyResolution.Unsupported : Automatic Runtime Dependency Resolution : Unsupported distribution/version '12'.`

Is there any way to bypass/override this version check so I could try the Ubuntu installer and see if it works?

Thanks!

Jesse

**Comment by graewert-biosaxs** [comment]
Hi Jesse,
I am sorry to hear about your trouble. Indeed, ATSAS packages were continued only if asked for by our customers and no one asked for Debian. To our knowledge, you are also the first academic asking for Debian. We have a meeting next Tuesday and get back to you after that.
Best regards, Tobias

**Reply by graewert-biosaxs** [reply]
Hi Jesse,

We have discussed the matter:
- we need to keep version checking.
- it is possible to set up a build environment for debian and so provide you with a debian version. I hate to bring it to you, we can only offer this service if you pay for it. Could you please send me an email to confirm?

**Comment by fmneto** [comment]
Hello

Could you clarify on why the installation is locked to certain distributions? I'm not sure why version checking based on distribution is so important. 

Without some clarification that seems arbitrary and only makes everything needlessly difficult for users and for you as well. 

This comment was originally meant to ask for help since I'm running Debian too, but after re-reading your Dec 11 response I have given up on ATSAS for the time being.

**Reply by franke-biosaxs** [reply]
We can not support all distributions under the sun - at least not as long as we are not compensated for the effort and time. BIOSAXS GmbH is a for-profit company that provides ATSAS for free to academic users as a service. If you would like to sponsor Debian support for the community, please do not hesitate to contact us.

**Reply by fmneto** [reply]
Thanks for replying but that doesn't explain _why_ locking the installation to certain distributions is a requirement. It actually sounds like the team went out of their way to create this limitation - using effort and time to impose an artificial restriction that can only hinder usage of the ATSAS package.

**Reply by franke-biosaxs** [reply]
"The team" is, for the most part, me. And for context, see my other message to Maxim:
> There is a reason why those checks are there. People used to install binaries of distribution A on distribution B - and then started to complain when issues arose of those, e.g. missing libraries, mismatching library versions etc.

Unexplained crashes are often part of this. Many bug reports simply say ’it doesn’t work,’ and identifying the real cause—in this case mismatching distributions—takes more time than enforcing strict compatibility checks (been there, done that).

To be clear: we only support distributions also used by our paying commercial customers. The community benefits from their support, as it enables us to offer free academic versions at all. That said, we’re open to supporting other distributions—provided someone is willing to sponsor the effort. While we get occasional requests, no one has stepped forward so far.

**Reply by fmneto** [reply]
That makes everything crystal clear. There is nothing left to be said.

**Comment by maxim-biosaxs** [comment]
As a not quite professional suggestion:

One may install on Ubuntu first and then copy the files over to Debian.
This seems to work. @franke-biosaxs : Would it be bad idea?

**Reply by franke-biosaxs** [reply]
Hi Maxim. Yes. There is a reason why those checks are there. People used to install binaries of distribution A on distribution B - and then started to complain when issues arose of those, e.g. missing libraries, mismatching library versions etc.

**Comment by jbhopkins** [comment]
Since I started this thread, I'm going to pop back in and comment. I had a fruitful offline exchange with @graewert-biosaxs, and while we're not in a position to sponsor a Debian build at this time I understand the reasoning, and appreciate that there continue to be free for academics versions of ATSAS. 

I'm not sure what the protocol is for these new discussion forums, I'm fine if this thread is closed but will leave it open for the moment and let the moderators decide.

**Reply by franke-biosaxs** [reply]
Hi Jesse. It's an open discussion forum. I think we keep it open in case anyone else might want to chime in, one way or another.

**Comment by ng5511** [comment]
Hi, I hope it is fine if I jump back onto this issue again. I am trying to install the Ubuntu-24 Version on Linux Mint 22 (which is based on Ubuntu 24). I circumvented the error described in the original post by using the option to install the dependencies manually. However, I was so far unable to find a list of the dependencies to make sure I have them installed on my system. Can I find those somehwere?

Best,
Niels

**Reply by franke-biosaxs** [reply]
Hi Niels. 

Indeed, they are not listed in the install description. Below should be the complete list of packages that would be installed, taken from the installer (Ubuntu names). Otherwise you are on your own with this. Enjoy the ride ;)

> libc6, libgcc1, libquadmath0, libstdc++6, libxml2, liblzma5, libgfortran5, libgomp1, libbrotli1, libicu74, libtiff6

**Reply by ng5511** [reply]
Hi, thanks for the super quick reply! At the end I circumvented the problem by setting up a VM running Ubuntu and installing ATSAS there. Probably also a viable solution for anyone reading this thread this thread in the future ;)
